src/generators/node-config-schema/constants.mjs

```diff
@@ -0,0 +1,51 @@
+// Error Messages
+export const ERRORS = {
+  missingCCandHFiles:
+    'Both node_options.cc and node_options.h must be provided.',
+  headerTypeNotFound:
+    'Header type for "{{headerKey}}" not found in the header file.',
+  missingTypeDefinition:
+    'No type definition found for header type "{{headerType}}" in TYPE_DEFINITION_MAP.',
+};
+
+// Regex pattern to match calls to the AddOption function.
+export const ADD_OPTION_REGEX =
+  /AddOption[\s\n\r]*\([\s\n\r]*"([^"]+)"(.*?)\);/gs;
+
+// Regex pattern to match header keys in the Options class.
+export const OPTION_HEADER_KEY_REGEX = /Options::(\w+)/;
+
+// Basic JSON schema for node.config.json
+export const BASIC_SCHEMA = {
```

Comment:
Is this constant should be external ? I mean is it can be pass as arg.

Because now if there change on core we need to update this repo. And ideally we want to not touch to this repo

Reply:
It's extremely rare that these types change. The last time a new type was introduced was 7 years ago (https://github.com/nodejs/node/commit/29a71bae40ffa0bbc8ba6b2bdf051a09987da7f7).

If and when a new type is added, it's just a few lines of code to update this repository.

Reply:
I think the json schema should be put in this repository since there isn't really anything necessitating it to be in Node core